src/main/java/org/truffleruby/core/thread/ThreadNodes.java

```diff
@@ -376,6 +382,56 @@ public DynamicObject pass() {
 
     }
 
+    @CoreMethod(names = "safe_level")
+    public abstract static class SafeLevelNode extends CoreMethodArrayArgumentsNode {
+
+        @Child private ReadObjectFieldNode readNode;
+
+        @Specialization
+        public int safeLevel(DynamicObject self) {
+            return (int)getReadNode().execute(Layouts.THREAD.getThreadLocals(self));
```

Comment:
Missing space after `(int)` and I would rename `self` to `thread`.

Reply:
Fixed.